starfish/image/_filter/match_histograms.py

```diff
@@ -0,0 +1,124 @@
+from functools import partial
+from typing import Optional, Set, Union
+
+import numpy as np
+import xarray as xr
+
+from starfish.compat import match_histograms
+from starfish.imagestack.imagestack import ImageStack
+from starfish.types import Axes
+from starfish.util import click
+from ._base import FilterAlgorithmBase
+
+
+class MatchHistograms(FilterAlgorithmBase):
+
+    def __init__(
+            self, group_by: Set[Axes]
+    ) -> None:
+        """Normalize data by matching distributions of each tile or volume to a reference volume
+
+        Chunks sharing the same values for axes specified by group_by will be quantile
+        normalized such that their intensity values are identically distributed. The reference
+        distribution is calculated by sorting the intensities in each chunk and averaging across
+        chunks.
+
+        For example, if group_by={Axes.CH, Axes.ROUND} each (z, y, x) volume will be linearized,
+        the intensities will be sorted, and averaged across {Axes.CH, Axes.ROUND}, equalizing
+        the intensity distribution of each (round, channel) volume.
+
+        Setting group_by={Axes.CH} would carry out the same approach, but the result would _retain_
+        variation across channel.
+
+        Parameters
+        ----------
+        group_by : Set[Axes]
+        """
+        self.group_by = group_by
+
+    _DEFAULT_TESTING_PARAMETERS = {"group_by": {Axes.CH, Axes.ROUND}}
+
+    @staticmethod
+    def _harmonize_enum(iterable):
```

Comment:
general comment: this is plausibly a useful method for other places.

```suggestion
    def _harmonize_enum(iterable: Iterable[Axes]) -> Iterable[str]:
```